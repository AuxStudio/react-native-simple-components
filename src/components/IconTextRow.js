import React from "react";
import { View, Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import styleConstants from "../assets/styleConstants";

import Icon from "react-native-vector-icons/MaterialIcons";

export default function IconTextRow(props) {
    /*
    static get propTypes() {
        return {
            iconName: PropTypes.string,
            text: PropTypes.string,

            iconStyle: PropTypes.node,
            textStyle: PropTypes.node,
            style: PropTypes.node,
        };
    }
*/

    return (
        <View style={[styles.container, props.style]}>
            <View style={styles.iconContainer}>
                <Icon
                    name={props.iconName}
                    style={[styles.icon, props.iconStyle]}
                />
            </View>
            <View style={styles.textContainer}>
                <Text style={[styles.text, props.textStyle]}>{props.text}</Text>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        justifyContent: "space-between",
        borderBottomWidth: 1,
        borderBottomColor: styleConstants.dividerColor,
        paddingVertical: 16,
    },
    iconContainer: {},
    icon: {
        fontSize: styleConstants.iconFont,
        color: styleConstants.primaryText,
    },
    textContainer: {},
    text: {
        fontSize: styleConstants.regularFont,
        color: styleConstants.primaryText,
    },
});
