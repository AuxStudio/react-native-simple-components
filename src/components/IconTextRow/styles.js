import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderBottomColor: styleConstants.colors.dividerColor,
    paddingVertical: styleConstants.dimensions.padding.large,
  },
  icon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.primaryText,
  },
  text: {
    flex: 1,
    fontSize: styleConstants.fonts.sizes.regular,
    color: styleConstants.colors.primaryText,
    textAlign: 'right',
  },
});

export default styles;
