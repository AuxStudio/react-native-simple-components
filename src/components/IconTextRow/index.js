import IconTextRow from './IconTextRow';

export default IconTextRow;
