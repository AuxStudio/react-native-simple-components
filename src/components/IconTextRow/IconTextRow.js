import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';
import styleConstants from '../assets/styleConstants';

import Icon from 'react-native-vector-icons/MaterialIcons';

export default function IconTextRow(props) {
  /*
    static get propTypes() {
        return {
            iconName: PropTypes.string,
            text: PropTypes.string,

            iconStyle: PropTypes.node,
            textStyle: PropTypes.node,
            style: PropTypes.node,
        };
    }
*/

  return (
    <View style={[styles.container, props.style]}>
      <View style={styles.iconContainer}>
        <Icon name={props.iconName} style={[styles.icon, props.iconStyle]} />
      </View>
      <View style={styles.textContainer}>
        <Text style={[styles.text, props.textStyle]}>{props.text}</Text>
      </View>
    </View>
  );
}
