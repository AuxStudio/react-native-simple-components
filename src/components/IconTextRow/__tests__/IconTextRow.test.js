import React from 'react';
import renderer from 'react-test-renderer';
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';

import IconTextRow from '../';

describe('IconTextRow', () => {
  it('renders with all props', () => {
    expect(
      renderer.create(
        <IconTextRow
          iconName="check"
          text="Test"
          handlePress={jest.fn()}
          androidRipple
          androidRippleColor="red"
          androidRippleBorderless
          iconStyle={{ color: 'red' }}
          textStyle={{ color: 'red' }}
          style={{ backgroundColor: 'blue' }}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<IconTextRow />)).toMatchSnapshot();
  });

  it('renders with custom icon', () => {
    expect(
      renderer.create(<IconTextRow customIcon={<MaterialIcon name="check" />} />),
    ).toMatchSnapshot();
  });
});
