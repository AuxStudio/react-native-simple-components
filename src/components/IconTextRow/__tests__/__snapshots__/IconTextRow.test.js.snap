// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconTextRow renders with all props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "blue",
      "borderBottomColor": "#E0E0E0",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#212121",
            "fontSize": 24,
          },
          Object {
            "color": "red",
          },
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#212121",
          "flex": 1,
          "fontSize": 17,
          "textAlign": "right",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`IconTextRow renders with custom icon 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#E0E0E0",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#212121",
          "flex": 1,
          "fontSize": 17,
          "textAlign": "right",
        },
        undefined,
      ]
    }
  >
    IconTextRow
  </Text>
</View>
`;

exports[`IconTextRow renders with minimum required props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#E0E0E0",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#212121",
            "fontSize": 24,
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#212121",
          "flex": 1,
          "fontSize": 17,
          "textAlign": "right",
        },
        undefined,
      ]
    }
  >
    IconTextRow
  </Text>
</View>
`;
